### PR TITLE
Feature: Onboarding Form

### LINKED ISSUE #9: Onboarding Form
**Is your feature request related to a problem? Please describe.**
Need a way to input financial data to generate projections

**Describe the solution you'd like**
Form with fields:
- Current age (number)
- Retirement age (number)
- Annual income (currency)
- Current savings (currency)
- Monthly contribution (currency)
- Risk tolerance (dropdown: conservative / moderate / aggressive)
- Retirement goal (currency, optional)
- Account allocation (TFSA / RRSP / FHSA priority or percentages)

Submit -> POST /api/plans -> redirect to dashboard

**Describe alternatives you've considered**
Multi-step wizard is possible, but outside of the scope of this project.

**Additional context**
Ensure client-side validation and have helpful labels / tooltips for fields

### PR DESCRIPTION
**Pull request by mpolisch**
## Summary
Implements the onboarding flow for  new users through /onboarding, which is a multi-selection plan creation form that collects financial details and creates the users first retirement plan. Also adds the NavBar, CurrencyInput UI component, shared inputClassName style constant, plan/projection types, and frontend Zod schemas for plan creation.

## Approach & Design Decisions
- onboarding is a protected route and was added to proxy.ts. New users with no plans get redirected to /onboarding, while returning users are redirected to the dashboard (or whatever page they were previously on)
- hasFetchedRef in (protected)/layout.tsx: prevents duplicate /plans API calls on navigations. Plans check runs once per session, subsequent route changes use ref to skip fetch and unblock the layout directly
- routeReady gate means children don't get rendered in (protected) until routing is confirmed.
- startTransition on setRouteReady allows for a smoother transition in (protected) redirects 
- z.preprocess on optional fields to coerce empty strings to undefined before Zod validation runs, preventing z.coerce.number() from converting "" to 0 and failing .positive() on an intentionally empty optional field
- z.input + z.output split for CreatePlanInput and CreatePlanDTO, resolving a type conflict between z.coerce and react-hook-form's resolver
- CurrencyInput component adds a reusable $ prefix number input in components/ui/ and blocks e/E/+/- keys
- inputClassName extracted to styles.ts since its shared across loginform, registerform, and onboarding form
- contribution priority defaults to tfsa_first, mirroring the DB default

## Security Considerations
- All financial inputs validated client-side with Zod
- Cross-field validations are also considered
- e/E/+/- blocking prevents invalid numeric input
- Backend re-validates all inputs
- /onboarding protected by both proxy.ts and ProtectedLayout

## Related Issues
Closes #9 

## Testing
**What's covered:**
- [ ] Unit tests
- [x] Edge cases (list below)

**Edge cases considered:**
- Empty name preprocessed to undefined and DB defaults to 'My Retirement Plan'
- Empty retirement_goal procprocessed to undefined and stored as null
- current_age >= retirement_age triggers an inline Zod error
- monthly_contributions * 12 > annual income also triggers inline Zod error
- Account balances exceed total savings triggers inline Zod error
- Field below minimum or maximums (current_age & retirement_age) triggers inline Zod error
- New user login redirects to /onboarding
- Returning user login redirects to /dashboard
- Returning user manually navigating to /onboarding redirects to /dashboard
- Successful submission creates plan in DB and redirects to /dashboard

## Checklist
- [x] No secrets or credentials in code
- [x] Input validation in place
- [x] Error handling covers failure paths
- [x] Code is self-documenting or commented where non-obvious
- [x] No dead code or debug logs left in

## Notes
- hasFetchedRef is never reset between navigations within the session, only runs once per mount. If a user creates a plan in another tab, they'd need to refresh to see the updated routing behaviour

### CODE REVIEW

File: lib/schemas.ts [hunk 1]
```diff
@@ -20,9 +21,81 @@ export const registerSchema = z
...
     email: z.string().trim().toLowerCase().pipe(z.email('Invalid email format')),
     password: z.string().min(1, 'Password is required'),
 });
 
 export type LoginDTO = z.infer<typeof loginSchema>;
+
+// Base Plan Schema, shares properties between Create and Update
+export const planBaseSchema = z.object({
+    name: z.preprocess(
+        (val) => (val === '' || val === null || val === undefined ? undefined : val),
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply lib/schemas.ts]: fixed

File: lib/schemas.ts [hunk 2]
```diff
@@ -20,9 +21,81 @@ export const registerSchema = z
...
+        z.coerce.number().positive('Retirement goal must be positive').optional(),
+    ),
+    tfsa_balance: z.coerce.number().min(0, 'TFSA balance cannot be negative'),
+    rrsp_balance: z.coerce.number().min(0, 'RRSP balance cannot be negative'),
+    fhsa_balance: z.coerce.number().min(0, 'FHSA balance cannot be negative'),
+    contribution_priority: z
+        .enum(['tfsa_first', 'balanced', 'rrsp_heavy'], {
+            error: 'Invalid contribution priority',
+        })
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply lib/schemas.ts]: removed error codes entirely since input is a select with a default

File: lib/schemas.ts [hunk 3]
```diff
@@ -20,9 +21,81 @@ export const registerSchema = z
...
+        .number()
+        .int('Age must be a whole number')
+        .min(30, 'Retirement age must be at least 30')
+        .max(90, 'Retirement age must be 90 or under'),
+    annual_income: z.coerce.number().positive('Annual income must be positive'),
+    current_savings: z.coerce.number().min(0, 'Current savings cannot be negative'),
+    monthly_contributions: z.coerce.number().positive('Monthly contributions must be positive'),
+    risk_tolerance: z.enum(['conservative', 'moderate', 'aggressive'], {
+        error: 'Invalid risk tolerance',
+    }),
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply lib/schemas.ts]: removed error codes since input is a select with a default value

File: lib/schemas.ts [hunk 4]
```diff
@@ -20,9 +21,81 @@ export const registerSchema = z
...
+            code: 'custom',
+            message: 'Sum of accounts cannot exceed total savings',
+            path: ['current_savings'],
+        });
+    }
+});
+
+// Update Plan Schema
+export const updatePlanSchema = planBaseSchema.partial();
+
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply lib/schemas.ts:93]: doesn't matter since its update endpoint is a PUT

File: app/(protected)/layout.tsx [hunk 5]
```diff
@@ -24,5 +62,10 @@ export default function ProtectedLayout({ children }: { children: React.ReactNod
 
     if (!user) return null;
 
-    return <>{children}</>;
+    return (
+        <>
+            <NavBar />
+            {children}
+        </>
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply app/(protected)/layout.tsx]: fixed

File: app/(protected)/layout.tsx [hunk 6]
```diff
@@ -1,20 +1,58 @@
...
+
+        // After a redirect, pathname changes and this effect re-runs.
+        // hasFetchedRef prevents a second API call
+        if (hasFetchedRef.current) {
+            if (pathname === '/onboarding') {
+                router.replace('/dashboard');
+            } else {
+                startTransition(() => setRouteReady(true));
+            }
+            return;
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply app/(protected)/layout.tsx]: fixed
